libs/s25main/ai/aijh/AIPlayerJH.cpp

```diff
@@ -2346,59 +2346,45 @@ unsigned AIPlayerJH::GetNumAIRelevantSeaIds() const
-                        1 :
-                        0;
-    // Rollingpin
-    toolsettings[10] = (inventory.goods[GoodType::Rollingpin] + inventory.people[Job::Baker]
-                        < bldPlanner->GetNumBuildings(BuildingType::Bakery) + 1) ?
-                         1 :
-                         0;
-    // Shovel
-    toolsettings[5] = (toolsettings[4] < 1 && toolsettings[3] < 1 && toolsettings[6] < 1 && toolsettings[1] < 1
-                       && (inventory.goods[GoodType::Shovel] < 1)) ?
-                        1 :
-                        0;
-    // Axe
-    toolsettings[2] = (toolsettings[4] < 1 && toolsettings[3] < 1 && toolsettings[6] < 1 && toolsettings[1] < 1
-                       && (inventory.goods[GoodType::Axe] + inventory.people[Job::Woodcutter] < 12)
-                       && inventory.goods[GoodType::Axe] < 1) ?
-                        1 :
-                        0;
-    // Tongs(metalworks)
-    toolsettings[0] =
-      0; //(toolsettings[4]<1&&toolsettings[3]<1&&toolsettings[6]<1&&toolsettings[2]<1&&(aii.GetInventory().goods[GoodType::Tongs]<1))?1:0;
-    // cleaver
-    toolsettings[9] =
+    toolsettings[Tool::Saw] = (inventory.goods[GoodType::Saw] + inventory.people[Job::Carpenter] < 2) ?
+                                4 :
+                                inventory.goods[GoodType::Saw] < 1 ? 1 : 0;
+    toolsettings[Tool::PickAxe] = (inventory.goods[GoodType::PickAxe] < 1) ? 1 : 0;
+    toolsettings[Tool::Hammer] = (inventory.goods[GoodType::Hammer] < 1) ? 1 : 0;
+    toolsettings[Tool::Crucible] = (inventory.goods[GoodType::Crucible] + inventory.people[Job::IronFounder]
+                                    < bldPlanner->GetNumBuildings(BuildingType::Ironsmelter) + 1) ?
+                                     1 :
+                                     0;
+    // Only if we haven't ordered any basic tool, we may order other tools
+    const bool hasBasicToolsOrder = toolsettings[Tool::PickAxe] != 0 || toolsettings[Tool::Saw] != 0
+                                    || toolsettings[Tool::Crucible] != 0 || toolsettings[Tool::Hammer] != 0;
+    toolsettings[Tool::Scythe] = (!hasBasicToolsOrder && inventory.goods[GoodType::Scythe] < 1) ? 1 : 0;
+    toolsettings[Tool::Rollingpin] = (inventory.goods[GoodType::Rollingpin] + inventory.people[Job::Baker]
+                                      < bldPlanner->GetNumBuildings(BuildingType::Bakery) + 1) ?
+                                       1 :
+                                       0;
+    toolsettings[Tool::Shovel] = (!hasBasicToolsOrder && inventory.goods[GoodType::Shovel] < 1) ? 1 : 0;
+    toolsettings[Tool::Axe] =
+      (!hasBasicToolsOrder && inventory.goods[GoodType::Axe] + inventory.people[Job::Woodcutter] < 12
+       && inventory.goods[GoodType::Axe] < 1) ?
+        1 :
+        0;
+    toolsettings[Tool::Tongs] = 0; //(!hasBasicToolsOrder&&(aii.GetInventory().goods[GoodType::Tongs]<1))?1:0;
```

Comment:
so it never orders these? could it be that this is the "bug" why the AI does not produce any gold or weapons anymore at some point?

Reply:
Might be. I improved this by using a smart algorithm taking the buildings into account which need those tools